src/components/ui/InfiniteList/InfiniteList.stories.tsx

```diff
@@ -0,0 +1,71 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useState, useEffect, useRef } from 'react';
+
+import Flex from '../Flex';
+import InfiniteList from './';
+import InfiniteListDocs from './InfiniteList.mdx';
+
+export default {
+  title: 'UI Components/InfiniteList',
+  parameters: {
+    docs: {
+      page: InfiniteListDocs.parameters.docs.page().props.children.type
+    }
+  },
+  component: InfiniteList
+};
+
+export const BasicInfiniteList = () => {
+  // All of the below code is just to fake an API call that would return a new batch of messages
+  const [batchNumber, setBatchNumber] = useState(1);
+  const numberOfBatches = 5;
+  const batchSize = 50;
+  const batchRef = useRef(batchNumber);
+
+  const createBatch = () => {
+    if (batchRef.current <= numberOfBatches) {
+      const batch = [];
+      for (let i = 1; i <= batchSize; i++) {
+        batch.push(
+          <div style={{'border': '1px solid #3f4149', 'textAlign': 'center', 'backgroundColor': '#075fff', 'color': 'white', 'padding': '0.5rem'}}>
+            {((batchSize * (numberOfBatches - batchRef.current) + 1) + i - 1).toString()}
+          </div>
+        );
+      }
+      return batch
+    } else {
+      console.log("No more messages to send.")
+      return []
+    }
+  };
+
+  const [items, setItems] = useState([...createBatch()]);
+  const itemsRef = useRef(items);
+
+  useEffect(() => {
+      itemsRef.current = items;
+      batchRef.current = batchNumber;
+  });
```

Comment:
Is there any need for this to be in an effect?

Reply:
I think so. In order to keep these values up to date in real time, they need to be updated continually. If we try to just reference state, the data becomes stale.

Reply:
I was more asking if we need this logic in the effect, vs just in the render function directly. This is running on every render anyways, so I was wondering if there was a reason it needed explicitly to be in an effect

Since you mention it though, where is the data going stale? I don't see us creating an closures, so I'm wondering where we're needing these refs instead of just state